Code of conduct revision

arokem opened the pull request:
Also includes proposed procedures for report and enforcement of the CoC.

Based on the following Google Doc: https://docs.google.com/document/d/1Cj9trFifGNqyNtpyEO9FfJLlmDGTbUy5enRAA9rTyH4/edit?tab=t.0#heading=h.214z66dv3mpo.

As this still includes portions that need to be filled out and some issues that need to be ironed out (e.g., retention of records about enforcement actions), this is still considered a draft.

arokem committed:
Code of conduct rewrite.

Also includes proposed procedures for report and enforcement of the CoC.

Based on the following Google Doc: https://docs.google.com/document/d/1Cj9trFifGNqyNtpyEO9FfJLlmDGTbUy5enRAA9rTyH4/edit?tab=t.0#heading=h.214z66dv3mpo.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

arokem committed:
Teach codespell about the word "recuse".

arokem committed:
Merge branch 'coc-amendment' of github.com:arokem/bids-dot-github into coc-amendment

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,10 +1,83 @@
+
+[Reference](#reference)
+
+## Code of Conduct {#coc}
```
Style point: These `{#anchor}`s are being shown but not linked. This does not seem to be a part of GitHub-flavored markdown.

There is an outline available on all Markdown documents by clicking the button in the top right of the page:

![image](https://github.com/user-attachments/assets/8e003f4a-5691-4ecd-bb37-fd84463aefc1)

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -20,30 +93,295 @@ Examples of unacceptable behavior by participants include:
+
+### Follow up with the reporter {#follow-up-with-the-reporter}
+
+A person who makes a report should receive a follow-up email stating what action was taken in response to the report. If the work group decided no response was needed, they should provide an email explaining why it was not a Code of Conduct violation. Reports that are not made in good faith (such as "reverse sexism" or "reverse racism") may receive no response.
```
```suggestion
A person who makes a report should receive a follow-up email stating what action was taken in response to the report. If the work group decided no response was needed, they should provide an email explaining why it was not a Code of Conduct violation. Reports that are determined not to have been made in good faith may receive no response.
```

I really don't think it's a good idea to pick two particular positions to represent bad faith reports. These phrases have particular meanings in American political discourse, but that meaning may not be shared by people of all international backgrounds or people who have not had their ear to the ground in the last decade or so.

I think stating that the group has discretion to determine bad faith or disparate impact ("Impact" section above) is sufficient to cover these cases without inviting debate over what these terms mean, which debate could itself turn into CoC violations.

effigies reviewed (approved):
Overall LGTM. One suggestion, though I'm not going to argue it further if others disagree.

arokem committed:
Remove specifying language for bad faith examples, to be more portable.

arokem committed:
Merge branch 'coc-amendment' of github.com:arokem/bids-dot-github into coc-amendment

arokem committed:
Use better language.

arokem committed:
Remove toc and anchors.

arokem committed:
Update contact information for maintainers/steering.

arokem commented:
Apologies, this has taken me an inexcusably long time to return to. 

I addressed comments/suggestions made by @effigies above (thanks!). 

@bids-standard/steering and @bids-standard/maintainers : any further comments? 

The one point that I think still needs to be addressed, before this can be enacted is the text regarding previous members of the steering group also being available to contact. On the one hand, I think that having previous steering group members available can be helpful. On the other hand, I don't want to assume that everyone who has been on SG is now willing to serve in this capacity. 

But: I also think that adding/removing names from this document is potentially an issue as well. So, it would be good if we had some way to point to people who are interested in serving in this capacity on some other page in a similar way to how we currently point to maintainers and SG in the section titled "General BIDS reporting procedure". However, I am not sure how we should do that. A separate section for "CoC Working Group" adjacent to these contact pages may be a potential way to go.

effigies commented:
I think the response group is well-defined (3 people, at least one steering and one maintainer, unless conflicts of interest make this impossible), but the actions to be taken seem to flip between "working" and response groups, and it's a bit hard to follow.

Who is the working group? Is it just a synonym for response group, or does it mean the set of contactable people, i.e., current steering, current maintainers, and past steering who volunteer to remain in it?

arokem committed:
Clarify who the working group is and what a response group is.

arokem commented:
Good point! I have clarified in the last change-set that "Working group" is maintainers + steering and "response group" is a smaller sub-set to evaluate a particular report. I think we'd also need to add some words [around here](https://bids.neuroimaging.io/collaboration/governance.html#other-working-groups)

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -20,30 +22,294 @@ Examples of unacceptable behavior by participants include:
+
+When you make a report to a member of the Code of Conduct working group, they will gather information about the incident according to the BIDS Procedure For Incident Response.
+
+After an incident responder takes the report, they will immediately establish a 3 person Code of Conduct Response Group with at least 1 (past or present) BIDS steering group members and 1 BIDS maintainer, unless there is a conflict of interest, in which case any non-interested parties will be contacted.
```
Reads oddly to me. Maybe this?

```suggestion
After an incident responder takes the report, they will immediately establish a 3 person Code of Conduct Response Group with at least 1 (past or present) BIDS steering group members and 1 BIDS maintainer, unless conflicts of interest prevent such an arrangement, in which case any disinterested party may be recruited.
```

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -20,30 +22,294 @@ Examples of unacceptable behavior by participants include:
+
+#### Following up with reporters
+
+Within one week of an incident report, a member of the response group, will follow up with the person who made the report and provided their contact information. The follow up may include:
```
```suggestion
Within one week of an incident report, a member of the response group will follow up with the reporter. The follow up may include:
```

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -20,30 +22,294 @@ Examples of unacceptable behavior by participants include:
+
+#### Conflict of Interest
+
+If an member of the response group has a conflict of interest for a report, they will recuse themselves from the discussion and handling of the incident. The incident documentation will not be available to them, and they will excuse themselves from any conversations involving handling the incident.
```
```suggestion
If a member of the working group has a conflict of interest for a report, they will recuse themselves from the discussion and handling of the incident. The incident documentation will not be available to them, and they will excuse themselves from any conversations involving handling the incident.
```

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -20,30 +22,294 @@ Examples of unacceptable behavior by participants include:
+
+The BIDS CoC Working Group will be required to respond back with whether they accept the recommended response to the report. If they disagree with the recommended response, they should provide a detailed response or additional context as to why they disagree. The Working Group is encouraged to respond within a week.
+
+In cases where the Working Group disagrees on the suggested resolution for a report, the BIDS Code of Conduct work group may choose to notify the BIDS steering group.
```
The working group includes the steering group.

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -20,30 +22,294 @@ Examples of unacceptable behavior by participants include:
+
+### Follow up with the reported person
+
+The BIDS Code of Conduct work group will work with online community administrators/moderators to draft a response to the reported person. The email should contain:
```
```suggestion
The BIDS Code of Conduct response group will work with online community administrators/moderators to draft a response to the reported person. The email should contain:
```

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -20,30 +22,294 @@ Examples of unacceptable behavior by participants include:
+
+### Decide further responses
+
+If the reported person provides additional context, the BIDS Code of Conduct work group may need to re-evaluate the behavioral modification plan and consequences.
```
```suggestion
If the reported person provides additional context, the BIDS Code of Conduct response group may need to re-evaluate the behavioral modification plan and consequences.
```

effigies commented on CODE_OF_CONDUCT.md:
```diff
@@ -20,30 +22,294 @@ Examples of unacceptable behavior by participants include:
+
+### Follow up with the reporter
+
+A person who makes a report should receive a follow-up email stating what action was taken in response to the report. If the work group decided no response was needed, they should provide an email explaining why it was not a Code of Conduct violation. Reports that are determined not to have been made in good faith may receive no response.
```
```suggestion
A person who makes a report should receive a follow-up email stating what action was taken in response to the report. If the response group decided no response was needed, they should provide an email explaining why it was not a Code of Conduct violation. Reports that are determined not to have been made in good faith may receive no response.
```

effigies reviewed (commented):
Minor cleanups.

The status of past steering members in the CoC WG is still a little unclear by my reading. I think it might make sense to formalize the CoC WG as consisting of: Current steering, maintainers, and past steering members who consent to remain on the WG. I think those past members should either be both potential initial contacts and response group candidates or else be neither.

Informally, I would suggest we refrain from involving past SG members unless they are the point of contact chosen by the reporter or there are difficulties pulling together a response group from active steering/maintainers.

To that effect, I think adding a section on the CoC WG to https://bids.neuroimaging.io/collaboration/governance.html would make sense. Until that's done, I think listing ex-steering group here is fine.
